bint.go

```diff
@@ -210,48 +210,160 @@ func parseBintFromU128(s []byte) (bool, bint, uint8, error) {
 		coef u128
 		prec uint8
 	)
-	for ; pos < width; pos++ {
-		if s[pos] == '.' {
+
+	if len(s[pos:]) <= 19 {
+		coef, prec, err = parseSmallToU128(s[pos:])
+	} else {
+		coef, prec, err = parseLargeToU128(s[pos:])
+	}
+
+	if err == ErrInvalidFormat {
```

Comment:
```suggestion
	if errors.Is(err, ErrInvalidFormat) {
```

because you might wrap error at some point, and it will break

Reply:
this is intentional actually. These parsing function shouldn't wrap the error. You can see the error is wrapped at the return right below